File: Extra - Estudos/1451 - TecladoQuebrado/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        while (sc.hasNextLine()) {
            String linha = sc.nextLine();
            char[] resultado = new char[linha.length()];
            int inicio = 0;
            int fim = 0;
            int pos = 0;
            
            for (int i = 0; i < linha.length(); i++) {
                char c = linha.charAt(i);
                
                if (c == '[') {
                    pos = inicio; // Move para o início
                } 
                else if (c == ']') {
                    pos = fim; // Move para o fim
                } 
                else {
                    // Desloca os caracteres para abrir espaço
                    for (int j = fim; j > pos; j--) {
                        resultado[j] = resultado[j-1];
                    }
                    resultado[pos] = c;
                    pos++;
                    fim++;
                }
            }
            
            System.out.println(new String(resultado, 0, fim));
        }
        
        sc.close();
    }
}

File: Extra - Estudos/1451 - TecladoQuebrado/main.c
#include <stdio.h>
#include <string.h>

int main() {
    char entrada[100];
    char resultado[100];
    
    while (fgets(entrada, 100, stdin) != NULL) {
        // Remove o '\n' se existir
        entrada[strcspn(entrada, "\n")] = '\0';
        
        int inicio = 0;
        int fim = 0;
        int pos = 0;
        int len = strlen(entrada);
        
        for (int i = 0; i < len; i++) {
            char c = entrada[i];
            
            if (c == '[') {
                pos = inicio; 
            } 
            else if (c == ']') {
                pos = fim; 
            } 
            else {
                for (int j = fim; j > pos; j--) {
                    resultado[j] = resultado[j-1];
                }
                resultado[pos] = c;
                pos++;
                fim++;
            }
        }
        
        resultado[fim] = '\0';
        printf("%s\n", resultado);
    }
    
    return 0;
}